### PR TITLE
Migrate further functions out of transform.py, add some unit tests.

### PR DESCRIPTION
**Pull request by owenlittlejohns**
## Description

This PR migrates a couple more functions out of `transform.py`. It also adds a couple of unit tests for 3 of the easier to digest functions.

## Jira Issue ID

N/A

## Local Test Steps

Following the previous few PRs - pull this branch, build it, run the HGA regression test suite against it.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* [x] `version.txt` and `CHANGE.md` updated if any service code is changed.
* [x] Tests added/updated and passing.
* ~~Documentation updated (if needed).~~

### CODE REVIEW

File: gdal_subsetter/transform.py [hunk 1]
```diff
@@ -935,20 +916,19 @@ def subset2(self, tiffile, outputfile, bbox=None, band=None, shapefile=None):
...
-                        shapefile, ref_ds, shapefile_out
-                    )
+            if bbox:
+                shapefile_out = box_to_shapefile(tiffile, bbox)
+                boxproj, _ = boxwrs84_boxproj(bbox, tiffile)
+            else:
+                shapefile_out = os.path.join(os.path.dirname(outputfile),
+                                             'tmpshapefile')
+                boxproj = shapefile_boxproj(shapefile, tiffile,
+                                            shapefile_out)
```
Comment (owenlittlejohns) [inline gdal_subsetter/transform.py:926]: The additional outputs from this function were either not being used (`proj`, `geometryname`) or were already defined prior to the function call. So I simplified the output from `shapefile_boxproj` to just return what was needed.

File: tests/unit/test_coordinate_utilities.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+from osgeo import gdal
+
+from gdal_subsetter.coordinate_utilities import (calc_coord_ij, calc_ij_coord,
+                                                 is_rotated_geotransform)
+
+
+class TestCoordinateUtilities(TestCase):
```
Comment (owenlittlejohns) [inline tests/unit/test_coordinate_utilities.py:10]: It's not a whole heap - but it's a start.

File: tests/unit/test_coordinate_utilities.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    def test_calc_coord_ij(self):
+        """ Ensure the expected row and column indices are retrieved for a
+            given geotransform, x coordinate and y coordinate.
+
+        """
+        self.assertTupleEqual(calc_coord_ij((10, 1, 0, 20, 0, 2), 20, 60),
+                              (10, 20))
+
+    def test_calc_ij_coord(self):
+        """ Ensure the expect values are retrieved for a given geotransform,
```
Comment (flamingbear) [inline tests/unit/test_coordinate_utilities.py:22]:
```suggestion
        """ Ensure the expected values are retrieved for a given geotransform,
```

**Review by flamingbear — APPROVED** [review]
Just one typo, everything else looks great.

**Pull request merged**